src/models.py

```
@@ -5,6 +5,13 @@
 from dotenv import load_dotenv
 from nanoid import generate as generate_nanoid
 from pgvector.sqlalchemy import Vector
+
+# Resolve embedding dimensions from config; fallback to 1536 for backward compat.
+try:
+    from .config import settings as _settings
+    _EMBEDDING_DIMS: int = _settings.VECTOR_STORE.DIMENSIONS  # type: ignore[union-attr]
+except Exception:
```

Ruff (0.15.12)

[warning] Do not catch blind exception: `Exception`

(BLE001)